lib/hsa/mcwamp_hsa.cpp

```diff
@@ -3171,6 +3197,11 @@ HSADispatch::setLaunchConfiguration(int dims, size_t *globalDims, size_t *localD
       workgroup_total_size = workgroup_size[0] * workgroup_size[1] * workgroup_size[2];
     }
 
+    if (kernel->wavefront_sgpr_count > 128 ||
+        kernel->workitem_vgpr_count*workgroup_total_size > 4*64*256) {
```

Comment:
Could you comment on what "4 * 64 * 256" calculates?  Is it   4 (SIMD/CU) * 64 (lane/SIMD) * 256 (vgpr/lane) ?

I think we need to round-up the group size (if it hasn't been done in other places) to the next wave size because the HW schedule as the wavefront granularity.  

I think we need to calculate how many waves a workgroup would produce and see whether we have enough SIMDs to fit them in.    I believe there are cases where if we only look at the total # of gprs on a CU, it might be fine but the workgoup may still not fit in a CU because the remaining gprs are spread across different SIMDs